### PR TITLE
Add hictkpy

### PR DESCRIPTION
**Pull request by robomics**
Add recipe for [paulsengroup/hictkpy](https://github.com/paulsengroup/hictkpy)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by robomics** [comment]
@BiocondaBot please fetch artifacts

**Comment by robomics** [comment]
I am not sure where does `import hictkpy.hictkpy` come from:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/conda/conda-bld/hictkpy_1691574443981/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.10/site-packages/hictkpy/__init__.py", line 1, in <module>
    from .hictkpy import File, is_cooler, is_hic_file, __version__
ModuleNotFoundError: No module named 'hictkpy.hictkpy'
Tests failed for hictkpy-0.0.1-py310h8dd5466_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
WARNING:conda_build.build:Tests failed for hictkpy-0.0.1-py310h8dd5466_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
TESTS FAILED: hictkpy-0.0.1-py310h8dd5466_0.tar.bz2
```

**Comment by robomics** [comment]
I think the issue is due to python version mismatch.
Libs inside the wheel are named like `hictkpy.cpython-311-x86_64-linux-gnu.so`, while the Python version installed by conda in CI (and also on a clean environment on my machine) is 3.10.

How does one build one conda package for each python version supported by a  library (Python 3.6 and newer in this case)?

File: recipes/hictkpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{% set name = "hictkpy" %}
+{% set version = "0.0.1" %}
+{% set sha256 = "af87e7670447331e8e9f3f59f00a97de640c111317fd76b87d756a93fe0d9d47" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+build:
+  number: 0
```
Comment (mencian) [inline recipes/hictkpy/meta.yaml:10]:
```suggestion
  number: 0
  skip: True  # [py < 36]
```

File: recipes/hictkpy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - conan >=2
+    - make
+    - eigen >=3.4
+    - fast_float >=5
+    - fmt >=10
+    - highfive >=2.7
+    - pybind11 >=2.11
+    - spdlog >=1.12
+  host:
+    - python >=3.6
```
Comment (mencian) [inline recipes/hictkpy/meta.yaml:30]:
```suggestion
    - python
```

File: recipes/hictkpy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    - hdf5 >=1.12
+    - libdeflate
+    - numpy
+    - pandas
+    - scipy
+
+  run:
+    - python >=3.6
```
Comment (mencian) [inline recipes/hictkpy/meta.yaml:38]:
```suggestion
    - python
```

**Review by mencian — CHANGES_REQUESTED** [review]
A few comments inline to have separate builds for each Python versions >=3.6

**Comment by robomics** [comment]
@BiocondaBot please fetch artifacts

**Comment by robomics** [comment]
@BiocondaBot please add label

@mencian many thanks for your comments!

**Review by mencian — APPROVED** [review]

**Pull request merged**